frame/contracts/src/migration.rs

```diff
@@ -15,7 +15,45 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-//! Migration framework for pallets.
+//! Multi-block Migration framework for pallet-contracts.
+//!
+//! This module allows us to define a migration as a sequence of [`MigrationStep`]s that can be
+//! executed across multiple blocks.
+//!
+//! # Usage
+//!
+//! A migration step is defined under `src/migration/vX.rs`, where `X` is the version number.
+//! For example, `vX.rs` defines a migration from version X to version X + 1.
+//!
+//! ## Example:
+//!
+//! To configure a migration to v12 for a runtime using v9 of `pallet-contracts` on the chain, you
+//! would set the `Migrations` trait as follows:
+//!
+//! ```rust
+//! use pallet_contracts::migration::{v10, v11, v12};
+//! type Migrations = (v10, v11, v12);
```

Comment:
Shouldn't this be `vX::Migration<T>`?
```suggestion
//! type Migrations = (v9::Migration<Runtime>, v10::Migration<Runtime>, v11::Migration<Runtime>);
```

Reply:
Ah right! Merged